shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -35,6 +39,21 @@ static FlutterLocale FlutterLocaleFromNSLocale(NSLocale* locale) {
   return flutterLocale;
 }
 
+// Returns the unresolved path of the executable for the current process.
+static std::optional<std::string> GetExecutablePath() {
+  // Determine the size of the buffer required.
+  uint32_t pathSize = 0;
+  if (_NSGetExecutablePath(nullptr, &pathSize) == 0) {
+    return std::nullopt;
+  }
+  std::string executablePath;
+  executablePath.resize(pathSize);
+  if (_NSGetExecutablePath(executablePath.data(), &pathSize) != 0) {
+    return std::nullopt;
+  }
+  return executablePath;
+}
```

Comment:
Any preference between this and `[[NSProcessInfo processInfo]  arguments]]` (etc.)?

Reply:
It seems like `arguments` is a closer match to the engine API we're passing it to, and it would be a bit easier to use (not having to do the double-call thing).

Reply:
I believe the behaviour as implemented is correct.

According to the [doc comment](https://api.flutter.dev/flutter/dart-io/Platform/executable.html), it's expected to be "The path of the executable used to run the script in this isolate."

The behaviour on the standalone Dart VM is as follows:
```dart
import 'dart:io';

main() {
  print('Platform.executable: ${Platform.executable}');
  print('Platform.resolvedExecutable: ${Platform.resolvedExecutable}');
}
```

Execution:
```
% ../out/host_debug_unopt/dart-sdk/bin/dart main.dart
Platform.executable: ../out/host_debug_unopt/dart-sdk/bin/dart
Platform.resolvedExecutable: /Users/cbracken/Developer/flutter/engine/src/out/host_debug_unopt/dart-sdk/bin/dart
```